packages/cosmic-swingset/src/chain-main.js

```diff
@@ -288,6 +288,12 @@ export default async function main(progname, args, { env, homedir, agcc }) {
   // instance, and we update the 'portNums.storage' value each time toSwingSet is called
   async function launchAndInitializeSwingSet(bootMsg) {
     // this object is used to store the mailbox state.
+    // TODO: Generalize StorageAPI caching for both makeChainStorage and buildCrankBuffer?
```

Comment:
The two caching layers have different lifetimes and ownership:
* the crankbuffer is owned by the kernel, invisible to the host application, and closed/flushed/written-back when the kernel decides to commit a crank, and abandoned when the kernel decides to abandon a crank
* the outer cache (the "block buffer") is owned by the host app, invisible to the kernel, commited when the host commits the rest of its state, and only abandoned if/when the host app is terminated

It'd be interesting to think about using nested DB transactions to hold both buffers directly in the DB, but 1: we'd need a different DB (LMDB only offers a single transaction), and 2: we'd have to rethink how we build the crankhash.